src/prototype_creep_resources.js

```diff
@@ -182,10 +182,8 @@ Creep.prototype.handleUpgrader = function() {
   say(this);
   this.spawnReplacement(1);
   var room = Game.rooms[this.room.name];
-  if (room.memory.attackTimer > 50 && room.controller.level > 6) {
-    if (room.controller.ticksToDowngrade > 10000) {
-      return true;
-    }
+  if ((room.energyAvailable < room.energyCapacityAvailable / 4 || (room.controller.level > 6 && room.memory.attackTimer > 50)) && room.controller.ticksToDowngrade > 10000) {
```

Comment:
I wouldn't do it like that, maybe in combination with the storage.

What is the problem that you are trying to solve?
E.g. if you have enough energy in the storage, the harvester is too slow (to small, no roads, ...)